Lib/http/server.py

```diff
@@ -976,9 +976,10 @@ def test(HandlerClass=BaseHTTPRequestHandler,
     ServerClass.address_family, addr = _get_best_family(bind, port)
     HandlerClass.protocol_version = protocol
 
-    if tls_cert:
-        server = ThreadingHTTPSServer(addr, HandlerClass, certfile=tls_cert,
-                                      keyfile=tls_key, password=tls_password)
+    # Make sure ServerClass is a class object, not a function instance
+    if isinstance(ServerClass, type) and issubclass(ServerClass, HTTPSServer):
```

Comment:
This check is invalid. If ServerClass is a function instance, we would always fall back to a non-TLS connection which is incorrect.